contracts/modules/TransferManager/MaximumVolumeTransferManager.sol

```diff
@@ -0,0 +1,296 @@
+    * @param _index representing the array index of the maximum volume transfer restriction
+    */
+    function removeMaximumTransferRestriction(uint256 _index) public withPerm(ADMIN){
+        require (maximumVolumeRestrictions.length != 0, "no existing restrictions");
+        require (maximumVolumeRestrictions.length > _index, "invalid index");
+
+        if(_index < (maximumVolumeRestrictions.length - 1)){
+            MaximumVolumeRestriction memory lastRestriction = maximumVolumeRestrictions[maximumVolumeRestrictions.length - 1];
+            maximumVolumeRestrictions[_index] = lastRestriction;
+        }
+
+        maximumVolumeRestrictions.length = (maximumVolumeRestrictions.length).sub(1);
+
+        emit RemoveMaximumVolumeRestricion(
+            msg.sender,
+            now
+        );
+    }
+
+    /**
+     * @notice Return the permissions flag that are associated with MaximumVolumeTransferManager
+     */
+    function getPermissions() public view returns(bytes32[]) {
+        bytes32[] memory allPermissions = new bytes32[](1);
+        allPermissions[0] = ADMIN;
+        return allPermissions;
+    }
+
+    /**
+    * @notice returns a count of all the added maximum volume restrictions
+    */
+    function getMaximumVolumeRestrictionsCount() public view returns(uint)  {
+        return maximumVolumeRestrictions.length;
+    }
+
+    /**
+    * @notice gets rolling interval period value in seconds
+    * @param _rollingPeriodInterval one of five RollingIntervals enum value
+    */
+    function _getPeriod(RollingIntervals _rollingPeriodInterval) internal pure returns(uint256){
+        uint256 period;
+        if(_rollingPeriodInterval == RollingIntervals.Daily){
+            period = 24 hours;
+        }else if(_rollingPeriodInterval == RollingIntervals.Weekly){
+            period = 7 * 24 hours;
+        }else if(_rollingPeriodInterval == RollingIntervals.Quarterly){
+            period = 3 * 30 * 24 hours;
+        }else if(_rollingPeriodInterval == RollingIntervals.Biannually){
+            period = 6 * 30 * 24 hours;
+        }else{
+            period = 365 * 24 hours;
+        }
+        return period;
+    }
+
+    /**
+    * @notice internal function for checking restriction parameters
+    * @param _startTime timestamp representing when this maximum volume restriction starts
+    * @param _endTime timestamp representing when this maximum volume restriction ends
+    * @param _rollingPeriodInterval interval period representing one of five RollingIntervals
+    */
+    function _checkRestrictionParams(
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriodInterval
+    ) internal view {
+        require(
+            uint256(RollingIntervals.Annually) >= _rollingPeriodInterval,
+            "invalid rolling period interval - should be < 5"
+        );
+        require(_startTime == 0 || _startTime >= now, 'invalid startTime');
```

Comment:
okay but accordingly I've added `_endTime > now` to prevent anyone from modifying expired restrictions or adding invalid ones.
```suggestion
        require(_startTime == 0 || _startTime >= now, 'invalid startTime');
        require(_endTime > 0 && _endTime > _startTime && _endTime > now, 'invalid endTime');
```